File: sdk/web/component/kenviro/controller/ExpertSystemController.js
var ExpertSystemController = function ExpertSystemController(cfg)
{
    let ctrl = {};

    ctrl.Find = function (id)
    {
        let args = [id];
        cfg.Connector.Submit(args, "ardvro/component/kenviro/ExpertSystem/Find", function (box)
        {
            onResponse != null ? onResponse(box == null ? null : box.Data) : null;
        });
    };

    ctrl.SearchExpertSystems = function (text, onResponse)
    {
        let args = [text];
        cfg.Connector.Submit(args, "ardvro/component/kenviro/ExpertSystem/SearchExpertSystems", function (box)
        {
            onResponse != null ? onResponse(box == null ? null : box.Data) : null;
        });
    };

    ctrl.SearchMatters = function (expertSystemId, text, onResponse)
    {
        let args = [expertSystemId, text];
        cfg.Connector.Submit(args, "ardvro/component/kenviro/ExpertSystem/SearchMatters", function (box)
        {
            onResponse != null ? onResponse(box == null ? null : box.Data) : null;
        });
    };

    ctrl.Compute = function (list, profileId, onResponse)
    {
        let args = {
            Criterias: list,
            pfeProfileId: profileId
        };
        cfg.Connector.Submit(args, "ardvro/component/kenviro/ExpertSystem/Compute", function (box)
        {
            onResponse != null ? onResponse(box == null ? null : box.Data) : null;
        });
    };

    ctrl.GetNextCriteria = function (inferenceData, onResponse)
    {
        cfg.Connector.Submit(inferenceData, "ardvro/component/kenviro/ExpertSystem/GetNextCriteria", function (box)
        {
            onResponse != null ? onResponse(box == null ? null : box.Data) : null;
        });
    };

    ctrl.Generate = function (data, onResponse)
    {
        cfg.Connector.Submit(data, "ardvro/component/kenviro/ExpertSystem/Generate", function (box)
        {
            onResponse != null ? onResponse(box == null ? null : box.Data) : null;
        });
    };

    ctrl.sp_EsLogAnswerCount = function (datefrom, dateto, offset, pagesize, callback)
    {
        let args = [datefrom, dateto, (offset * pagesize), pagesize];
        cfg.Connector.Submit(args, "ardvro/component/kenviro/ExpertSystem/sp_EsLogAnswerCount", function (box)
        {
            callback != null ? callback(box == null ? null : box.Data) : null;
        });
    };

    ctrl.sp_DssItemCriteriaUsageCount = function (datefrom, dateto, offset, pagesize, callback)
    {
        let args = [datefrom, dateto, (offset * pagesize), pagesize];
        cfg.Connector.Submit(args, "ardvro/component/kenviro/ExpertSystem/sp_DssItemCriteriaUsageCount", function (box)
        {
            callback != null ? callback(box == null ? null : box.Data) : null;
        });
    };

    ctrl.sp_EsLogAnswerCountSolutions = function (datefrom, dateto, offset, pagesize, callback)
    {
        let args = [datefrom, dateto, (offset * pagesize), pagesize];
        cfg.Connector.Submit(args, "ardvro/component/kenviro/ExpertSystem/sp_EsLogAnswerCountSolutions", function (box)
        {
            callback != null ? callback(box == null ? null : box.Data) : null;
        });
    };


    return ctrl;
};

File: sdk/.net/ardvro.sdk.net/component/kenviro/controller/ExpertSystemController.cs
namespace ardvro.sdk.net.component.kenviro.controller;

using ardvro.adapter.ais.ent;
using ardvro.adapter.ais.dto;
using ardvro.core.ext.connection;
using ardvro.core.lib.common;

public class ExpertSystemController
{
    private IConnection _connection;
    public ExpertSystemController(IConnection connection)
    {
        _connection = connection;
    }

    public void Find(int id, Action<aisEs> callback)
    {
        _connection.Submit(new Object[] { id }, "ardvro/component/kenviro/ExpertSystem/Find", callback);
    }

    public void SearchExpertSystems(String text, Action<List<aisEs>> callback)
    {
        _connection.Submit(new Object[] { text }, "ardvro/component/kenviro/ExpertSystem/SearchExpertSystems", callback);
    }

    public void SearchMatters(int expertSystemId, String text, Action<Inference> callback)
    {
        _connection.Submit(new Object[] { expertSystemId, text }, "ardvro/component/kenviro/ExpertSystem/SearchMatters", callback);
    }

    public void GetNextCriteria(Inference data, Action<Inference> callback)
    {
        _connection.Submit(data, "ardvro/component/kenviro/ExpertSystem/GetNextCriteria", callback);
    }

    public void Compute(List<Criteria> list, int profileId, Action<Inference> callback)
    {
        Inference data = new Inference();
        data.Criterias = list;
        _connection.Submit(data, "ardvro/component/kenviro/ExpertSystem/Compute", callback);
    }

    public void Generate(GenerateExpertSystemData data, Action<Inference> callback)
    {
        _connection.Submit(data, "ardvro/component/kenviro/ExpertSystem/Generate", callback);
    }

    public void sp_EsLogAnswerCount(DateTime datefrom, DateTime dateto, int offset, int pagesize, Action<List<Flexible>> callback)
    {
        _connection.Submit(new Object[] { datefrom, dateto, offset, pagesize }, "ardvro/component/kenviro/ExpertSystem/sp_EsLogAnswerCount", callback);
    }

    public void sp_DssItemCriteriaUsageCount(DateTime datefrom, DateTime dateto, int offset, int pagesize, Action<List<Flexible>> callback)
    {
        _connection.Submit(new Object[] { datefrom, dateto, offset, pagesize }, "ardvro/component/kenviro/ExpertSystem/sp_DssItemCriteriaUsageCount", callback);
    }

    public void sp_EsLogAnswerCountSolutions(DateTime datefrom, DateTime dateto, int offset, int pagesize, Action<List<Flexible>> callback)
    {
        _connection.Submit(new Object[] { datefrom, dateto, offset, pagesize }, "ardvro/component/kenviro/ExpertSystem/sp_EsLogAnswerCountSolutions", callback);
    }

}
